src/main/java/org/snomed/snowstorm/core/data/services/CodeSystemVersionService.java

```diff
@@ -0,0 +1,184 @@
+     * V => CodeSystemVersion.getDependantVersionEffectiveTime()
+     * */
+    private final Map<String, Integer> dependantVersionCache = new HashMap<>();
+
+    @Autowired
+    private CodeSystemService codeSystemService;
+
+    @Autowired
+    private BranchService branchService;
+
+    @Autowired
+    private ElasticsearchOperations elasticsearchOperations;
+
+    /**
+     * Populate cache with the version of SCT the latest version
+     * of the CodeSystem depends on.
+     *
+     * @param codeSystems The CodeSystems to process.
+     */
+    public void initDependantVersionCache(List<CodeSystem> codeSystems) {
+        for (CodeSystem codeSystem : codeSystems) {
+            CodeSystemVersion latestVersion = codeSystem.getLatestVersion();
+            if (latestVersion == null || SNOMEDCT.equals(latestVersion.getShortName())) {
+                continue;
+            }
+            this.doGetDependantVersionForCodeSystemVersion(latestVersion)
+                .ifPresent(dependantVersion -> dependantVersionCache.putIfAbsent(latestVersion.toString(), dependantVersion));
+        }
+    }
+
+    /**
+     * Clear cache.
+     */
+    public void clearCache() {
+        LOGGER.info("Clearing cache 'dependantVersionCache'.");
+        this.dependantVersionCache.clear();
+    }
+
+    /**
+     * Get the version of SCT the CodeSystemVersion depends on.
+     *
+     * @param codeSystemVersion The CodeSystemVersion to process.
+     * @return The version of SCT the CodeSystemVersion depends on.
+     */
+    public Optional<Integer> getDependantVersionForCodeSystemVersion(CodeSystemVersion codeSystemVersion) {
+        if (codeSystemVersion == null) {
+            return Optional.empty();
+        }
+
+        String shortName = codeSystemVersion.getShortName();
+        if (SNOMEDCT.equals(shortName)) {
+            return Optional.empty();
+        }
+
+        LOGGER.debug("Looking for {}'s dependency version.", shortName);
+        String codeSystemVersionString = codeSystemVersion.toString();
+        Integer dependantVersion = dependantVersionCache.get(codeSystemVersionString);
+        if (dependantVersion == null) {
+            LOGGER.debug("CodeSystemVersion '{}' not found in cache.", codeSystemVersionString);
+            Optional<Integer> optionalDependency = this.doGetDependantVersionForCodeSystemVersion(codeSystemVersion);
+            if (optionalDependency.isEmpty()) {
+                LOGGER.debug("No dependency version found for '{}'.", codeSystemVersionString);
+                return Optional.empty();
+            }
+
+            LOGGER.debug("Adding '{}' to cache.", codeSystemVersionString);
+            dependantVersion = optionalDependency.get();
+            dependantVersionCache.put(codeSystemVersionString, dependantVersion);
+        }
+
+        return Optional.of(dependantVersion);
+    }
+
+    private Optional<Integer> doGetDependantVersionForCodeSystemVersion(CodeSystemVersion codeSystemVersion) {
```

Comment:
The logic for actually working out the `dependantVersionEffectiveTime` was required in several places so I made it a shared, private method.